client-firewall-rules.md

```diff
@@ -3,38 +3,110 @@
+## Registry FQDN Endpoints
 
 To access MCR, the following FQDNs are required. 
 
 | Purpose | Protocol | Target FQDN | Available |
 | - | - | - | - |
 | Registry Endpoint | https | `mcr.microsoft.com` | Now |
-| Data Endpoint | https | `*.cdn.mscr.io` | Now - April 20, 2020 |
-| Data Endpoint | https | `*.data.mcr.microsoft.com` | Starting April  13, 2020 |
+| Data Endpoint | https | `*.cdn.mscr.io` | Now |
+| Data Endpoint | https | `*.data.mcr.microsoft.com` | Starting June  15, 2020 |
 
-If you're updating this prior to April 13, adding both date endpoints will assure a smooth transition. Once the consolidated data endpoint deployment is complete, we'll update this page.
+If you're updating this prior to June 15th, adding both data endpoints will assure a smooth transition. Once the consolidated data endpoint deployment is complete, we'll update this page enabling the removal of `*.cdn.mscr.io`.
 
 ![Azure Application Rule](./media/mcr-client-firewall-rules.png)
 
+## Testing the `*.cdn.mscr.io` Data Endpoint
+
+To validate configurations pre March 3rd, 2020, the following multi-arch images are available for testing. 
+
+### Windows & Linux
+To test pulling an image:  
+`docker pull mcr.microsoft.com/mcr/hello-world`
+
+Visualizing the data urls:  
+`curl https://mcr.microsoft.com/v2/hello-world/blobs/sha256:fce289e99eb9bca977dae136fbe2a82b6b7d4c372474c9235adc1741675f587e`
+
+produces a url with `[oldRegionReference].cdn.mscr.io` :
+```
+<a href="https://mcrwcus0.cdn.mscr.io/f439446e126348d29d8a8c39253da1ca-gqcvpppbjg//docker/registry/v2/blobs/...
+```
+
+## Testing the `*.data.mcr.microsoft.com` Data Endpoint
+
+To validate changes will work post March 3rd, 2020, the following images are available for testing today.
+
+**Note:** Separate repos for Windows and Linux are provided for verification:
+
+### Windows
+
+`docker pull mcr.microsoft.com/mcrlabscdnone/hello-world`
+
+Visualizing the data urls:  
+`curl  https://mcr.microsoft.com/v2/mcrlabscdnone/hello-world/blobs/sha256:fce289e99eb9bca977dae136fbe2a82b6b7d4c372474c9235adc1741675f587e`
+
+produces a url with `[region].data.mcr.microsoft.com` :
+```
+<a href="https://westcentralus.data.mcr.microsoft.com/f439446e126348d29d8a8c39253da1ca-gqcvpppbjg//docker/registry/v2/blobs/...
+```
+
+### Linux
+
+`docker pull mcr.microsoft.com/mcrlabscdnone/hello-world-linux`
+
+Visualizing the data urls:  
+`curl  https://mcr.microsoft.com/v2/mcrlabscdnone/hello-world-linux/blobs/sha256:fce289e99eb9bca977dae136fbe2a82b6b7d4c372474c9235adc1741675f587e`
+
+produces a url with `[region].data.mcr.microsoft.com` :
+```
+<a href="https://westcentralus.data.mcr.microsoft.com/f439446e126348d29d8a8c39253da1ca-gqcvpppbjg//docker/registry/v2/blobs/...
+```
+
+### Q&A
+- **Q: What do I do now, to prepare for this change?**  
+  **A:** Follow the above guidance, adding both data endpoints. 
```

Comment:
to your outbound firewall rules. If you do not use outbound firewall rules, no change is necessary.